### DISCUSSION TITLE
Cloudblock no longer working following instance reboot

### DISCUSSION THREAD

**Discussion by DavidBC71** [chadgeary/cloudblock#99] [Q&A]
Hi Chad, 

Thanks for this project. I've been using your cloudblock / pihole on a PC, Mac and iPhone without any problems for 18 months. It's been very useful for me. At the end of August there was an issue at the oracle datacentre and I received emails saying a reboot had been detected on my virtual machine instance. Soon after, further emails reported the problem had been resolved. Since then cloudblock / pihole havn't been working on either my PC, Mac or iPhone. Wireguard seems to connect to oracle (green tick appears and some uploaded data) but no internet connection. I've spent hours on zoom with oracle support but without a resolution. They claim everything on their end is working fine and can't suggest a solution. I guess some sort of setting has changed but I have no idea what. Can you suggest a fix? Is my best option to delete my oracle instances and start from scratch again? Thanks for your help. 

PS I can attach the event notification email if that would help.

**Comment by chadgeary** [comment]
Hi David, thank you

It's great to hear it has been working well for 18 months maintenance free.
I regret I am not as active as I'd like to be with this project due to
external obligations.

Here are a few troubleshooting tips that may help:

1. Connect to the oracle compute instance from the PC that ran terraform -
if this was windows please use the WSL environment:

option A (wireguard on):
```
ssh ubuntu@172.18.0.1
```

option B (public internet):
```
ssh ubuntu@<public-ip-of-compute-instance>
```

For both options, an SSH public key (created at install time) is used. The
key only exists on the computer that ran terraform. Option B only works if
your IP address has not changed. If your IP address has changed, you'll
need to re-run terraform apply after updating "mgmt_cidr" in oci.tfvars.

If a connection is successful, run the following commands:

```
sudo docker ps
sudo systemctl status cloudblock-ansible-state.service
```

these commands will give an overview of the cloudblock state

you can email me the output directly chad.geary [AT] gmail.com


2. Destroy (and re-create)

It's probably much simpler to re-deploy a new cloudblock, but oracle's free
tier does not provide enough resources to deploy multiple cloudblock
resources simultaneously
If the original PC still has WSL and the original cloudblock/oci/ folder
intact, try the following from command terminal:

```
terraform output
```

and read the two destroy instructions (remove the bucket objects first,
then terraform destroy)

If successful, review the values in oci.tfvars (ensure mgmt_cidr is
up-to-date, etc) and re-run

```
terraform apply -var-file="oci.tfvars"
```


3. Manual destroy or abandon (and re-create)

If the original PC no longer has the cloudblock/oci folder it may be
impossible to run terraform destroy as the state file doesn't exist. If
terraform apply was run again, it would deploy another set of resources
(and fail because of the free-tier limitation or name overlaps).
In this case, your two options are either:

option A (manually delete objects)
visit the oracle cloud web console and remove all of the cloudblock
objects, these should be prefixed with "cloudblock-", especially ensure the
virtual network, compute instance, and encryption vault(s) are removed

option B (abandon)
if unable to completely clean out the resources in the original oracle
account, it may be best to abandon/terminate the entire account and create
a new one. Oracle may be picky about creating another account so I suggest
using a different payment validation method (different card). Oracle's free
tier is still the best option, but there are some very good or low cost
alternatives. It could be worthwhile to try out azure or aws (they provide
a limited time credit after which the service becomes PAYG but is fairly
inexpensive). DigitalOcean is a flat $5/month if you're willing to swing it
and their cloud console is fairly simple.


Best of luck!

EDIT: fixed some formatting that was lost from email

**Comment by DavidBC71** [comment]
Hi Chad,

Thanks for prompt and comprehensive reply.

I'll give my results to your suggestions above

1. Option A, with Wireguard on

david@DESKTOP-8M51D2J:~$ ssh ubuntu@172.18.0.1
ssh: connect to host 172.18.0.1 port 22: Resource temporarily unavailable

1. Option B with Wireguard off

david@DESKTOP-8M51D2J:~$ ssh ubuntu@xxx.xxx.xx.xxx (where the x are my public IPv4 address)
ssh: connect to host 132.145.61.171 port 22: Resource temporarily unavailable

2.

david@DESKTOP-8M51D2J:~$ terraform output
╷
│ Warning: No outputs found
│
│ The state file either has no outputs defined, or all the defined outputs are empty. Please define an output in your
│ configuration with the `output` keyword and run `terraform refresh` for it to become available. If you are using
│ interpolation, please verify the interpolated value is not empty. You can use the `terraform console` command to
│ assist.
╵
david@DESKTOP-8M51D2J:~$ terraform refresh
╷
│ Warning: Empty or non-existent state
│
│ There are currently no remote objects tracked in the state, so there is nothing to refresh.

I haven't yet tried to manually delete, abandon and recreate, 3rd option.

After my initial deployment I did terraform destroy. I presume that is responsible for terraform commands not creating an output. I still have the original oci folders. As far as I know my IP address hasn't changed, using same machine, internet provider and other equipment.

Any ideas before I proceed to 3?

Thanks again for taking the time to help.

David

**Comment by DavidBC71** [comment]
Not sure why some text in previous answer has been scored out. Not done by me.

**Reply by fn5** [reply]
It's due to the ~ in your original pasted output. Not sure why it triggers all strikethrough, but remove it and it won't be there.
https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax
